core/src/main/java/org/eclipse/hono/config/AbstractConfig.java

```diff
@@ -54,6 +54,7 @@
     private FileFormat trustStoreFormat;
     private FileFormat keyFormat;
     private List<String> secureProtocols = Collections.singletonList("TLSv1.2");
+    private boolean sni;
```

Comment:
IMHO this should go into `ServerConfig` as it is only relevant/used for a server.